crates/bevy_ecs/src/storage/thin_array_ptr.rs

```diff
@@ -0,0 +1,307 @@
+use crate::query::DebugCheckedUnwrap;
+use std::alloc::{alloc, handle_alloc_error, realloc, Layout};
+use std::mem::needs_drop;
+use std::num::NonZeroUsize;
+use std::ptr::{self, NonNull};
+
+/// Similar to [`Vec<T>`], but with the capacity and length cut out for performance reasons.
```

Comment:
```suggestion
/// Similar to [`Vec<T>`], but with the capacity and length cut out for performance reasons.
///
/// This type can be treated as a `ManuallyDrop<Box<[T]>` without a built in length. To avoid,
/// memory leaks, [`drop`]Self::drop) must be called when no longer in use.
```